hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java

```diff
@@ -120,6 +121,45 @@ protected Map<EntityIDPair, Long> fungibleChangesFrom(
         return fungibleChanges;
     }
 
+    /**
+     * Gets all non-fungible tokenRelation balances for all modified token relations from the given {@link WritableTokenRelationStore}.
+     *
+     * @param writableTokenRelStore the {@link WritableTokenRelationStore} to get the token relation balances from
+     * @param tokenStore the {@link ReadableTokenStore} to get the token from
+     */
+    protected void nonFungibleChangesFrom(
```

Comment:
Hmm...
 - This is just a copy+paste of the `fungibleChangesFrom()` method with a different name, right?
 - And the goal is to make the record fungible transfer list include the #-of-NFT-owned "balances" that change **specifically** when a deleted or expired non-fungible token is dissociated from an account, correct?

Assuming I understand the goal here, imo instead we should:
 1. Rename `fungibleChangesFrom()` to `tokenBalanceChangesFrom()`; and,
 2. Add a third parameter to specify the `TokenType` that should be filtered for [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java#L98).

(Also, since this code will run **every** `handleTransaction`, we might consider adding a `boolean` return value to the `tokenBalanceChangesFrom()` method that lets us detect whether any token types were skipped and only do this in the specific case above.)

But this amount of duplication is undesirable imo. 😞

Reply:
Yes, you're right. I've refactor the change. Can you take a look again? Thank you 🙏